### PR TITLE
feat: --konnect-disable-consumer-sync flag

### LINKED ISSUE #6189: Add feature flag to skip consumer sync to Konnect
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Does this enhancement require public documentation?

- [X] I have added an Acceptance Criteria item for adding and/or adjusting public documentation (if applicable)

### Problem Statement

Some users have a lot of consumers. This can make Konnect sync very slow and cause dataplane sync issues (xref: https://github.com/Kong/kubernetes-ingress-controller/issues/6188 ).

Users are ok to not sync consumers as a workaround until Konnect provide a better solution for this.

### Proposed Solution

- add `--SkipConsumerKonnectSync` feature flag which bypasses consumers when syncing in Konnect

### Additional information

_No response_

### Acceptance Criteria

- [ ] Add feature flag and skip sync
- [ ] Validate that it works in Konnect
- [ ] Document this new flag

### PR DESCRIPTION
**Pull request by mlavacca**
**What this PR does / why we need it**:

**Which issue this PR fixes**:

Fixes #6189 

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: internal/adminapi/konnect.go [hunk 1]
```diff
@@ -28,6 +28,7 @@ type KonnectConfig struct {
 	LicenseSynchronizationEnabled bool
 	InitialLicensePollingPeriod   time.Duration
 	LicensePollingPeriod          time.Duration
+	DisableConsumersSync          bool
```
Comment (programmer04) [inline internal/adminapi/konnect.go:31]:
```suggestion
	ConsumersSyncDisabled      bool
```
Maybe like that to follow naming convention of other field `LicenseSynchronizationEnabled`?

**Review by programmer04 — COMMENTED** [review]
Should we put somewhere in which situation this option may be useful 🤔

**Comment by mlavacca** [comment]
> Should we put somewhere in which situation this option may be useful 🤔

In addition to the comment in the code, I've created an issue to add proper documentation.

**Review by randmonkey — COMMENTED** [review]
Should we add test cases for `ConsumerDisabled` is true?

File: internal/dataplane/kong_client.go [hunk 2]
```diff
@@ -740,6 +740,12 @@ func (c *KongClient) maybeSendOutToKonnectClient(
 		return nil
 	}
 
+	// In case users have many consumers, konnect sync can be very slow and cause dataplane sync issues.
+	// For this reason, if the --disable-consumers-sync flag is set, we do not send consumers to Konnect.
+	if konnectClient.ConsumersSyncDisabled() {
+		s.Consumers = nil
```
Comment (randmonkey) [inline internal/dataplane/kong_client.go:746]: Would it be better to set an option in `ToDeckContent` to control the generated config? modifying a shared pointer is quite dangerous in concurrent sending (would be required in #6188.)

Comment (mlavacca) [reply internal/dataplane/kong_client.go:746]: When we implement the concurrent sending, we'll have to duplicate the payload in my opinion, as the konnect kongstate can be different from the Gateway one. For example, if this flag is enabled, konnect's payload won't have any consumer, while the gateway's will.

**Review by programmer04 — APPROVED** [review]

**Pull request merged**